packages/wasm-parser/src/decoder.js

```diff
@@ -1752,12 +1752,17 @@ export function decode(ab: ArrayBuffer, opts: DecoderOpts): Program {
           t.sectionMetadata("custom", startOffset, sectionSizeInBytesNode)
         ];
 
+        const offsetBeforeReadUTF8 = offset;
         const sectionName = readUTF8String();
+        const bytesEatenByReadUTF8 = offset - offsetBeforeReadUTF8;
         eatBytes(sectionName.nextIndex);
```

Comment:
`sectionName.nextIndex` already contains the `bytesEatenByReadUTF8` information, it's the amount of bytes that we can eat after the read.